pkg/vadmin/fetch_node_state_vc.go

```diff
@@ -27,5 +30,48 @@ import (
 // or DOWN in our consensous state. It returns a map of vnode to its node state.
 func (v *VClusterOps) FetchNodeState(ctx context.Context, opts ...fetchnodestate.Option) (map[string]string, ctrl.Result, error) {
 	v.Log.Info("Starting vcluster FetchNodeState")
-	return nil, ctrl.Result{}, fmt.Errorf("not implemented")
+
+	// get fetch node state options
+	s := fetchnodestate.Parms{}
+	if err := s.Make(opts...); err != nil {
+		return nil, ctrl.Result{}, err
+	}
+
+	// call vcluster-ops library to fetch node states
+	vopts := v.genFetchNodeStateOptions(&s)
+	nodesInfo, err := v.VFetchNodeState(&vopts)
+	if err != nil {
+		v.Log.Error(err, "failed to fetch node states")
+		return nil, ctrl.Result{}, err
+	}
+
+	// parse node states
+	stateMap := map[string]string{} // node name to state map
+	for _, nodeInfo := range nodesInfo {
```

Comment:
We should only return information for the hosts that the caller was interested. We can use the `s.HostsNeeded` map for that.